### DISCUSSION TITLE
How should the private key be stored?

### DISCUSSION THREAD

**Discussion by fastovenko** [tejado/obsidian-gpgCrypt#21] [Q&A] [answered]
I saved the public and private keys in the Obsidian folder and used a passphrase. Where and how should I store the private key for security?

**Comment by oshev** [comment]
Yes, I don't understand this part either. If the private key is also saved in the vault, the security is compromised. Anyone who got access to the vault will have access to your private key. What's the point of encryption then?

**Reply by fastovenko** [reply]
I think the main point of this encryption is to use a password that no one knows.

**Reply by tejado** [reply]
It is now supported to store the keys with OpenPGP.js also outside the Vault: https://github.com/tejado/obsidian-gpgCrypt/releases/tag/0.4.0

**Comment by tejado** [comment, marked as answer]
There are multiple options, to explain the main ones:

1. Store the key pair within your Vault (e.g. to get it synchronized as well) and use a passphrase.
This makes setup dead‑simple and ensures your key follows you across every synced device and the private key is protected by the passphrase.

3. Store the key pair outside your Vault with or without a passphrase.
Keeping it off the synced folder reduces the blast radius if your vault is ever compromised.

4. Use GnuPG CLI Wrapper to store the key pair in GnuPG or on a smartcard (e.g. like a YubiKey).
This uses GnuPGs native keyring or hardware token so your private key never appears in plaintext on disk. There are lot of options with this, e.g. you can still protect your private key with a passphrase but this is then handled by GnuPG (e.g. more specific cache times). 
To store the private key then on a hardware token (Yubikey/smartcard) would add a physical‑presence requirement.
